### PR TITLE
(BOLT-98) Sudo support

### PR DESCRIPTION
**Pull request by pcarlisle**

### CODE REVIEW

File: lib/bolt/node/ssh.rb [hunk 1]
```diff
@@ -68,12 +78,22 @@ def execute(command, options = {})
           raise "could not execute command: #{command.inspect}" unless success
 
           channel.on_data do |_, data|
-            result_output.stdout << data
+            if use_sudo && data == sudo_prompt
+              channel.send_data "#{@password}\n"
```
Comment (joshcooper) [inline lib/bolt/node/ssh.rb]: Should this be `sudo_password`?

Comment (pcarlisle) [reply lib/bolt/node/ssh.rb]: yes, per TODO list in the pr description

**Comment by MikaelSmith** [comment]
Oops, BOLT-45 introduced a conflict.

**Comment by MikaelSmith** [comment]
A commit comment about how this works would also be really helpful...

File: lib/bolt/node/ssh.rb [hunk 2]
```diff
@@ -54,9 +54,19 @@ def disconnect
...
       status = {}
+      use_sudo = sudoable && (@sudo || @run_as)
+      sudo_prompt = '[sudo] Bolt needs to run as another user, password: '
+      if use_sudo
+        user_clause = if @run_as
+                        "-u #{@run_as}"
+                      else
+                        ''
+                      end
+        command = "sudo -S #{user_clause} -p '#{sudo_prompt}' #{command}"
```
Comment (MikaelSmith) [inline lib/bolt/node/ssh.rb]: Couldn't this use `"#{@sudo} ...` to make it slightly easier to add different sudo options in the future?

Comment (MikaelSmith) [reply lib/bolt/node/ssh.rb]: I guess I should just look at doing that, doesn't seem too hard to add a few.

File: lib/bolt/node/ssh.rb [hunk 3]
```diff
@@ -68,12 +78,22 @@ def execute(command, options = {})
           raise "could not execute command: #{command.inspect}" unless success
 
           channel.on_data do |_, data|
-            result_output.stdout << data
+            if use_sudo && data == sudo_prompt
```
Comment (MikaelSmith) [inline lib/bolt/node/ssh.rb]: How many times will this retry if the password is wrong?

Comment (pcarlisle) [reply lib/bolt/node/ssh.rb]: as many times as it's prompted

File: lib/bolt/node/ssh.rb (hunk 2 above)
Comment (pcarlisle) [reply lib/bolt/node/ssh.rb]: Almost every part needs to be parameterized to add another method so it didn't seem worth it to do it until we actually added one.

**Comment by MikaelSmith** [comment]
Ok, that merge attempt didn't work right.

**Pull request merged**